File: app/src/main/java/com/main/climbingdiary/fragments/StatisticFragment.java
package com.main.climbingdiary.fragments;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.content.ContextCompat;
import android.view.LayoutInflater;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

import com.main.climbingdiary.R;
import com.main.climbingdiary.chart.RouteBarChartController;
import com.main.climbingdiary.chart.RouteLineChartController;
import com.main.climbingdiary.controller.AppBarMenu;
import com.main.climbingdiary.controller.TableView;
import com.main.climbingdiary.models.Colors;
import com.main.climbingdiary.models.MenuValues;

import lombok.AccessLevel;
import lombok.NoArgsConstructor;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class StatisticFragment extends Fragment implements RouteFragment {

    private Button setLineChartBtn, setBarChartBtn, setTableBtn;
    private View view;

    private static StatisticFragment INSTANZ = null;

    public static StatisticFragment getInstance() {
        if (INSTANZ == null) {
            INSTANZ = new StatisticFragment();
        }
        return INSTANZ;
    }

    @Override
    public View getView() {
        return view;
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {

        view = inflater.inflate(R.layout.statistic_fragment, container, false);
        //set the bar barChart as Standard
        setBarChartBtn = view.findViewById(R.id.btn_stat);
        setLineChartBtn = view.findViewById(R.id.btn_dev);
        setTableBtn = view.findViewById(R.id.btn_table);
        //set the views
        final RouteLineChartController _routeLineChart = new RouteLineChartController(view);
        final TableView _routeTable = new TableView(view.getContext(), view);
        final RouteBarChartController _routeBarChartController = new RouteBarChartController(view);
        //the Button Click Listener
        setLineChartBtn.setOnClickListener(v -> {
            resetButtonBackground();
            setLineChartBtn.setBackgroundColor(Colors.getActiveColor());
            _routeLineChart.show();
            _routeTable.hide();
            _routeBarChartController.hide();
            _routeLineChart.createChart();
        });
        setBarChartBtn.setOnClickListener(v -> {
            resetButtonBackground();
            setBarChartBtn.setBackgroundColor(Colors.getActiveColor());
            _routeTable.hide();
            _routeLineChart.hide();
            _routeBarChartController.show();
            _routeBarChartController.createChart();
        });
        setTableBtn.setOnClickListener(v -> {
            resetButtonBackground();
            setTableBtn.setBackgroundColor(Colors.getActiveColor());
            _routeBarChartController.hide();
            _routeLineChart.hide();
            _routeTable.show();
            _routeTable.createTableView();
        });
        //default visualisation
        setBarChartBtn.callOnClick();
        return view;
    }

    //XAxis Formatter
    public void resetButtonBackground() {
        setLineChartBtn.setBackgroundColor(ContextCompat.getColor(view.getContext(), R.color.buttonColor));
        setBarChartBtn.setBackgroundColor(ContextCompat.getColor(view.getContext(), R.color.buttonColor));
        setTableBtn.setBackgroundColor(ContextCompat.getColor(view.getContext(), R.color.buttonColor));
    }

    @Override
    public void refreshData() {
        try {
            RouteLineChartController _routeLineChart = new RouteLineChartController(view);
            TableView _routeTable = new TableView(view.getContext(), view);
            RouteBarChartController _routeBarChartController = new RouteBarChartController(view);
            _routeBarChartController.createChart();
            _routeLineChart.createChart();
            _routeTable.createTableView();
        } catch (Exception ignored) {
        }
    }

    @Override
    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
        // Do something that differs the Activity's menu here
        super.onCreateOptionsMenu(menu, inflater);
        AppBarMenu appmenu = new AppBarMenu(menu);
        appmenu.setItemVisebility(MenuValues.FILTER, true);
        appmenu.setItemVisebility(MenuValues.SEARCH, false);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        item.setChecked(true);
        if (item.getGroupId() == R.id.filter_area + 1) {
            return true;
        }
        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/com/main/climbingdiary/fragments/StatisticFragment.kt
package com.main.climbingdiary.fragments

import android.content.res.ColorStateList
import android.os.Bundle
import android.view.*
import android.widget.Button
import androidx.core.content.ContextCompat
import androidx.fragment.app.Fragment
import com.main.climbingdiary.R
import com.main.climbingdiary.chart.RouteBarChartController
import com.main.climbingdiary.chart.RouteLineChartController
import com.main.climbingdiary.controller.AppBarMenu
import com.main.climbingdiary.chart.TableView
import com.main.climbingdiary.models.MenuValues

class StatisticFragment: Fragment(), RouteFragment {

    private lateinit var setLineChartBtn: Button
    private lateinit var setBarChartBtn: Button
    private lateinit var setTableBtn: Button
    private lateinit var view: View
    private lateinit var routeLineChartController: RouteLineChartController

    override fun getView(): View {
        return view
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        view = inflater.inflate(R.layout.statistic_fragment, container, false)
        //set the bar barChart as Standard
        setBarChartBtn = view.findViewById(R.id.btn_stat)
        setLineChartBtn = view.findViewById<Button>(R.id.btn_dev)
        setTableBtn = view.findViewById(R.id.btn_table)
        //set the views
        routeLineChartController = RouteLineChartController(view)
        val tableView = TableView(view.context, view)
        val routeBarChartController = RouteBarChartController(view)
        //the Button Click Listener
        setLineChartBtn.setOnClickListener {
            resetButtonBackground()
            setButtonActive(setLineChartBtn)
            routeLineChartController.show()
            tableView.hide()
            routeBarChartController.hide()
            routeLineChartController.createChart()
        }
        setBarChartBtn.setOnClickListener {
            resetButtonBackground()
            setButtonActive(setBarChartBtn)
            tableView.hide()
            routeLineChartController.hide()
            routeBarChartController.show()
            routeBarChartController.createChart()
        }
        setTableBtn.setOnClickListener {
            resetButtonBackground()
            setButtonActive(setTableBtn)
            routeBarChartController.hide()
            routeLineChartController.hide()
            tableView.show()
            tableView.createTableView()
        }
        //default visualisation
        setBarChartBtn.callOnClick()
        return view
    }

    //XAxis Formatter
    fun resetButtonBackground() {
        val textColor = ContextCompat.getColor(view.context, R.color.colorPrimaryDark)
        val background = ColorStateList.valueOf(
            ContextCompat.getColor(
                view.context,
                R.color.backgroundColorDark
            )
        )
        listOf(setLineChartBtn, setBarChartBtn, setTableBtn).forEach {
            it.backgroundTintList = background
            it.setTextColor(textColor)
        }
    }

    private fun setButtonActive(button: Button) {
        button.backgroundTintList = ColorStateList.valueOf(
            ContextCompat.getColor(view.context, R.color.colorAccent)
        )
        button.setTextColor(ContextCompat.getColor(view.context, R.color.white))
    }

    override fun refreshData() {
        try {
            val routeLineChart = RouteLineChartController(view)
            val routeTable = TableView(view.context, view)
            val routeBarChartController = RouteBarChartController(view)
            routeBarChartController.createChart()
            routeLineChart.createChart()
            routeTable.createTableView()
        } catch (ignored: Exception) {
        }
    }

    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
        // Do something that differs the Activity's menu here
        super.onCreateOptionsMenu(menu, inflater)
        val appmenu = AppBarMenu(menu)
        appmenu.setItemVisebility(MenuValues.FILTER, true)
        appmenu.setItemVisebility(MenuValues.SEARCH, false)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        item.isChecked = true
        return if (item.groupId == R.id.filter_area + 1) {
            true
        } else super.onOptionsItemSelected(item)
    }
}
